receiver/k8slogreceiver/README.md

```diff
@@ -0,0 +1,173 @@
+# K8slog Receiver
+
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [development]: logs   |
+| Distributions | [] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Areceiver%2Fk8slog%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Areceiver%2Fk8slog) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Areceiver%2Fk8slog%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Areceiver%2Fk8slog) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@h0cheung](https://www.github.com/h0cheung), [@TylerHelmuth](https://www.github.com/TylerHelmuth) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+<!-- end autogenerated section -->
+
+Tails and parses logs in k8s environment.
+
+There only one mode of discovery as for now, it's specified by the `discovery.mode` configuration option:
```

Comment:
Is `discovery.mode` the right term? What discovery takes place? Could it just be `mode`? Could it be you'd want to get all of those modes at once?

Reply:
Before submitting this PR, the configuration was discussed between the leader, co-workers, and myself. We feel that the top level configuration should be similar to `filelogreceiver` so that users can use it more easily. And the new feature to discover targets from k8s pods and containers will have a configuration under the `discovery` key. So I used `discovery.mode`. Other ideas are welcome.

The `discovery` means that otel will scan all k8s pods running on its node and find logs from them according to the user's configuration. After `discovery`, this will be a filelog-like component that will collect the log entries.

After both `daemonset-stdout` and `daemonset-file` are implemented, there may be a mixed mode of them, but this should not happen for `sidecar`.